sirepo/package_data/static/js/sirepo-plotting.js

```diff
@@ -3052,992 +3071,110 @@ SIREPO.app.directive('particle', function(plotting, layoutService, utilities) {
+    this.extrema = function(pArr, dim, reverse) {
+        var sPArr = self.sortInDimension(pArr, dim, reverse);
+        if(! sPArr) {
+            return null;
+        }
+        return sPArr.filter(function (point) {
+            return point[dim] == sPArr[0][dim];
+        });
+    };
 
-                $scope.yAxisTop = vpTop[1]/2.0;
-                $scope.xAxisLeft = vpLeft[0]/2.0;
-
-                //srdbg('box in vp L:', vpLeftX, 'r:', vpRightX, 'b:', vpBottomY, 't:', vpTopY, 'w:', vpWidth, 'h:', vpHeight);
-                //srdbg('box in vp b:', vpBottom, 'bi:', vpBottomIn, 'th:', theta, 'tanth:', tanTheta);
-
-                var camPos = cam.getPosition();
-                var vtkCanvas;
-                if(firstRender) {
-                    vtkCanvas = $($element).find('.sr-plot-particle-3d canvas')[0];
-                    //srdbg('ctx', vtkCanvas.getContext('webgl2'));
-                    //srdbg('found canvas', vtkCanvas, $(vtkCanvas).width(), $(vtkCanvas).height());
-                    //var w = $(vtkCanvas).attr('width');
-                    //$(vtkCanvas).attr('height', ASPECT_RATIO * w);
-                    //$(vtkCanvas)
-                    /*
-                    viewPlane = vtk.Common.DataModel.vtkPlane.newInstance({
-                        origin: [camPos[0], camPos[1], camPos[2]],
-                        normal: [0,0,1]
-                    });
-                    */
-                    firstRender = false;
-                }
-
-                select('.vtk-canvas-holder svg')
-                    .attr('width', $scope.width + $scope.margin.left + $scope.margin.right)
-                    .attr('height', $scope.height + $scope.margin.top + $scope.margin.bottom);
-
-                // domain is the value of the data points
-                // range is the position on the screen
-                // TODO (mvk): plotAxis should handle arbitrary rotated axes instead of doing it here
-                //axes.x.scale.range([0, $scope.vtkCanvasGeometry().size.width - 96]);
-                axes.x.scale.range([0, vpWidth]);
-                //axes.y.scale.range([$scope.vtkCanvasGeometry().size.height - 360, 0]);
-                axes.y.scale.range([vpHeight, 0]);
-                axes.z.scale.range([0, 180]);
-                $.each(axes, function(dim, axis) {
-                    axis.updateLabelAndTicks({
-                        width: $scope.vtkCanvasGeometry().size.width,
-                        height: $scope.vtkCanvasGeometry().size.height
-                    }, select);
+    // Returns the members of an array of edges (point pairs) that
+    // contain any of the points in another array
+    this.edgesWithCorners = function(edgeArr, pArr) {
+        var edges = edgeArr.filter(function (edge) {
+            return edge.some(function (corner) {
+                return pArr.some(function (point) {
+                    return point[0] == corner[0] && point[1] == corner[1];
```

Comment:
I like naming things: isEqual? Maybe even have a point object so you can just call a function on it with an argument.